packages/drivers/routerlicious-driver/src/tokens.ts

```diff
@@ -44,4 +44,12 @@ export interface ITokenProvider {
      * whether token came from cache.
      */
     fetchStorageToken(tenantId: string, documentId: string, refresh?: boolean): Promise<ITokenResponse>;
+
+    /**
+     * A callback triggered directly after creating the document.
+     * @param documentId - Document ID.
+     * @param creationToken - A special token that doesn't provide any kind of access, but it has the user's payload
+     * and document id. It can be used to validate the identity of the document creator.
+     */
+    documentPostCreateCallback?(documentId: string, creationToken: string): Promise<void>;
```

Comment:
Can you please add to description why it's an async task, and what driver does (or does not) while awaiting it? I.e. is the creation process stalled while waiting for it? Is it expected that this callback will throw exceptions if something goes wrong and how the rest of pipeline will behave? (I.e. is file deleted in such case?) What's the purpose of this interaction (if it's not fire-and-forget)?

Reply:
+1 to noting that this will slow down the document creation process